kasa/smart/modules/firmware.py

```diff
@@ -162,6 +155,7 @@ async def get_update_state(self) -> DownloadState:
         state = resp["get_fw_download_state"]
         return DownloadState(**state)
 
+    @update_after
```

Comment:
Perhaps separate the `@update_after` handling into its own PR and keep this one just for deferred updates?

Reply:
As per discord keeping this but renamed `@allow_update_after`